src/custom/state/lists/hooks/hooksMod.ts

```diff
@@ -21,140 +12,52 @@ import {
-      return tokenMap
-    }
-    if (!tokenMap[token.chainId]) tokenMap[token.chainId] = {}
-    tokenMap[token.chainId][token.address] = {
-      token,
-      list,
-    }
-    return tokenMap
-  }, {}) as TokenAddressMap
-    listCache?.set(list, map)
-    return map
-  }
-*/
-
-  const map = list.tokens.reduce<TokenAddressMap>((tokenMap, tokenInfo) => {
-    const token = new WrappedTokenInfo(tokenInfo, list)
-    const tokensByNetwork = tokenMap[token.chainId] || {}
-    if (tokensByNetwork[token.address] !== undefined) {
-      console.error(new Error(`Duplicate token! ${token.address}`))
-      return tokenMap
-    }
-    return {
-      ...tokenMap,
-      [token.chainId]: {
-        ...tokensByNetwork,
-        [token.address]: {
-          token,
-          list,
-        },
-      },
-    }
-  }, {})
-  listCache?.set(list, map)
-  return map
-}
-
-// export function useAllLists(): AppState['lists']['byUrl'] {
 export function useAllLists(): AppState['lists'][ChainId]['byUrl'] {
-  // MOD: adds { chainId } support to the hooks
   const { chainId: connectedChainId } = useWeb3React()
   const chainId = supportedChainId(connectedChainId) ?? DEFAULT_NETWORK_FOR_LISTS
-  // return useAppSelector<AppState, AppState['lists']['byUrl']>(state => state.lists.byUrl)
+
   return useAppSelector((state) => state.lists[chainId]?.byUrl, shallowEqual)
 }
 
-/**
- * Combine the tokens in map2 with the tokens on map1, where tokens on map1 take precedence
- * @param map1 the base token map
- * @param map2 the map of additioanl tokens to add to the base map
- */
-/* export function combineMaps(map1: TokenAddressMap, map2: TokenAddressMap): TokenAddressMap {
-  const chainIds = Object.keys(
-    Object.keys(map1)
-      .concat(Object.keys(map2))
-      .reduce<{ [chainId: string]: true }>((memo, value) => {
-        memo[value] = true
-        return memo
-      }, {})
-  ).map((id) => parseInt(id))
-
-  return chainIds.reduce<Mutable<TokenAddressMap>>((memo, chainId) => {
-    memo[chainId] = {
-      ...map2[chainId],
-      // map1 takes precedence
-      ...map1[chainId],
-    }
-    return memo
-  }, {}) as TokenAddressMap
-} */
-
-// merge tokens contained within lists from urls
-export function useCombinedTokenMapFromUrls(urls: string[] | undefined): TokenAddressMap {
-  // MOD: recreated here to allow it to use the custom useAllLists hook
+export function useTokensListFromUrls(urls: string[] | undefined): TokenInfo[] {
   const lists = useAllLists()
+
   return useMemo(() => {
-    if (!urls) return EMPTY_LIST
+    if (!urls) return []
 
     return (
       urls
         .slice()
         // sort by priority so top priority goes last
         .sort(sortByListPriority)
-        .reduce((allTokens, currentUrl) => {
-          const current = lists?.[currentUrl]?.current
-          if (!current) return allTokens
-          try {
-            const newTokens = Object.assign(listToTokenMap(current))
```

Comment:
Got rid of (at least 6) redundant cycles:
 - listToTokenMap (1 cycle)
 - combineMaps (5 cycles)